### PR TITLE
CRM-16806: Validate supplied entity type.

### PR DESCRIPTION
**Pull request by xurizaemon**
---
- [CRM-16806](https://issues.civicrm.org/jira/browse/CRM-16806)

### CODE REVIEW

File: CRM/Admin/Page/APIExplorer.php [hunk 1]
```diff
@@ -105,7 +105,9 @@ public static function getExampleFile() {
    * Ajax callback to display code docs
    */
   public static function getDoc() {
-    if (!empty($_GET['entity']) && strpos($_GET['entity'], '.') === FALSE) {
+    // Verify the API handler we're talking to is valid.
+    $entities = civicrm_api3('Entity', 'get');
+    if (!empty($_GET['entity']) && in_array($_GET['entity'], $entities['values']) && strpos($_GET['entity'], '.') === FALSE) {
       $entity = _civicrm_api_get_camel_name($_GET['entity']);
```
Comment (colemanw) [inline CRM/Admin/Page/APIExplorer.php]: This line can be deleted now, since we verified above that it's already in CamelCase.

**Comment by colemanw** [comment]
@xurizaemon what do you think of suggestion above?

**Comment by colemanw** [comment]
Added it in #6475

**Comment by eileenmcnaughton** [comment]
Per comments on the ticket I have not worried about porting this to 4.4- it does not apply cleanly & may not even be relevant

**Pull request merged**